pyccel/codegen/printing/ccode.py

```diff
@@ -1201,7 +1201,7 @@ def _print_Return(self, expr):
             # make sure that stmt contains one assign node.
             assert(len(last_assign)==1)
             variables = last_assign[0].rhs.get_attribute_nodes(Variable, excluded_nodes=(FunctionDef,))
-            unneeded_var = not any(b not in vars_in_deallocate_nodes for b in variables)
+            unneeded_var = not any(b not in vars_in_deallocate_nodes and not b.is_argument for b in variables)
```

Comment:
Why do you need this? Are we deallocating arguments?!

Reply:
i'm sorry, the problem was in the first attempt to fix, it should  `unneeded_var = not any(b in vars_in_deallocate_nodes for b in variables)`